include/boost/beast/core/detail/buffer.hpp

```diff
@@ -77,6 +78,132 @@ dynamic_buffer_prepare(
 #endif
 }
 
+inline
+net::mutable_buffer&
+truncate(net::mutable_buffer& target, std::size_t n)
+{
+    n = (std::min)(n, target.size());
+    target = net::mutable_buffer(target.data(), n);
+    return target;
```

Comment:
This whole function could be reduced to
```
return net::mutable_buffer( target.data(), (std::min)(n, target.size() );
```
At which point why do we need a whole separate function?